src/App.tsx

```diff
@@ -129,6 +136,45 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    public consolidateAllValues(): Values {
```

Comment:
Let's create a new function called `mapMerge` in a new file perhaps called `map-utils.ts`

```js
function mapMerge<K, V>(a: Map<K, V>, b: Map<K, V>): Map<K, V>
```

Note how mapMerge is generic and can be used combining maps of any type.

Also be careful on explicitly defining the behavior when both maps contain the same key (which map gets precedence). Note this behavior in the jsdoc.

Reply:
On second thought, perhaps `mapMerge` can take in a variable number of arguments: https://www.damirscorner.com/blog/posts/20180216-VariableNumberOfArgumentsInTypescript.html

Reply:
I would have to change the way that permanent values work. It is currently an array and not a Map. Plus I don't think I'd ever need a mapMarge function. I think `new Map([...map1, ...map2, ...map3])` works well.

Reply:
You should change the way permanent values work to be consistent with the way `Values` works. Logically they are the same type.